### PR TITLE
Add ReceiveData test and general listener

### PR DESCRIPTION
**Pull request by icraggs-sparkplug**
This is a working (host application) receive data test - I used Ignition to verify it.

However, I don't see that there are corresponding assertions in the spec to add to this test.  Is this true?  As section 5.9, Data Publish is currently empty, then that might explain it?  

As this test is for receipt of edge and device data messages, the assertions would be on the Host Application side, such as "when the host application receives a data message from an edge node or device, the corresponding data in the host application should be updated ASAP", or similar.  What do you think?

### CODE REVIEW

**Comment by wes-johnson** [comment]
> As this test is for receipt of edge and device data messages, the assertions would be on the Host Application side, such as "when the host application receives a data message from an edge node or device, the corresponding data in the host application should be updated ASAP", or similar. What do you think?

I agree with this. I'll look to add some of that before our next meeting. Did you want to go ahead and merge what you already have or wait? I'll leave that up to you.

**Review by wes-johnson — APPROVED** [review]

**Comment by icraggs-sparkplug** [comment]
I can't merge it myself, as I'm not actually a committer on the project - we didn't finish that off - but I haven't needed to be to open issues or PRs. 

I'd like this to be merged, so we can make other PRs based on it.

**Pull request merged**
